### ISSUE TITLE
[TextFields] Publish accessibility guidance in readme

### ISSUE THREAD

**Issue by jmdetloff** [material-components/material-components-ios#4325]
Guidance on how clients should handle MDCTextField accessibility should be added to its readme.

**Comment by romoore** [comment]
Need to know which properties/values have to be configured by the client.

**Comment by randallli** [comment]
@jmdetloff do you know off hand which properties need to be configured?

**Comment by romoore** [comment]
I think this is a larger conversation we can take offline (or put in a separate issue somewhere). I haven't given it much thought in general, nor have I played around with TextFields' accessibilty much.  Regarding how Text Fields work, I can (at the moment) think of a few guiding principles:

1.   **Do what `UITextField` does.** As much as practicable, we should match the behavior of `UITextField` to minimize the surprise of both users and developers.  Where we do deviate from `UITextField`, we should try to match the themes/concepts as much as possible.
2.  **Prefer usability over cross-component consistency.** While it's important that our components behave similarly have similar functionality or styling (minimizing surprise), I feel it's more important for each component to be as usable as possible.  For text fields, that could mean ensuring that the primary use cases are as easy as possible, even if that means deviating from how another component (like Cards) behaves.
3.   Focus on the primary use cases, even at the (potential) expense of secondary use cases. Secondary use cases should not be burdensome to use, but we shouldn't be optimizing for them either.  If we have real data telling us what the actual primary/secondary use cases (by frequency) are, we should probably defer to that.  Here are the primary use cases I can think of:
     * Entering text into an empty text field without errors.
     * Entering text into an empty text field and generating an error.
     * Editing existing text in a text field with an error.

**Comment by andrewoverton** [comment]
This is really solid general advice, thanks @romoore! I looked into it more and UITextField behaves similarly enough that I'm not sure anything needs to change in MDCTextField. I think I know what to write 👍
